OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -90,7 +92,7 @@ public void Tick(Squad owner)
 				}
 			}
 
-			var leader = owner.Units.ClosestTo(owner.TargetActor.CenterPosition);
+			var leader = owner.Units.ClosestToWithPathTo(owner.TargetActor);
```

Comment:
I think this line may fix #17327, by using pathfinding distance to the target

Reply:
As for the other part, we need to limit leader's locomotor or use other method to prevent entire squad stuck on terrain.

Reply:
This is still using the geometric distance to determine closeness, so it won't aid with that sadly.

Reply:
Have filed #20228 to cover it though.